### PR TITLE
WW-11: delete sighting endpoint

### PR DESCRIPTION
**Pull request by Palak843**
We have made a Delete Sighting endpoint.

### CODE REVIEW

File: backend/Repositories/SightingRepo.cs [hunk 1]
```diff
@@ -24,5 +27,25 @@ public IEnumerable<Sighting> GetAllSightings()
...
                 .Include(s => s.Species);
         }
+
+         public Sighting GetById(int id)
+        {
+            return _context
+                .Sightings
+                .Single(s => s.Id == id);
+        }
+         public Sighting DeleteSighting(int id)
```
Comment (TimLeach635) [inline backend/Repositories/SightingRepo.cs:37]: Line separating these two methods please!
```suggestion
        }
        
         public Sighting DeleteSighting(int id)
```

Comment (TimLeach635) [inline backend/Repositories/SightingRepo.cs:37]: I'm not sure how it happened but the methods are indented by one space more than the brackets!
```suggestion
        public Sighting GetById(int id)
        {
            return _context
                .Sightings
                .Single(s => s.Id == id);
        }
        public Sighting DeleteSighting(int id)
```

Comment (TimLeach635) [inline backend/Repositories/SightingRepo.cs:37]: You don't need to call it `DeleteSighting` - it's the Sightings Repo, we know what we're deleting! `DeleteById` is probably a better name (and is consistent with your other method, `GetById`)

File: backend/Repositories/SightingRepo.cs [hunk 2]
```diff
@@ -24,5 +27,25 @@ public IEnumerable<Sighting> GetAllSightings()
...
+        {
+             Sighting sighting = _context
+                .Sightings
+                .Include(s => s.Species)
+                .Single(s => s.Id == id);
+            sighting.IsDeleted = true;
+            _context.Sightings.Update(sighting);
+            _context.SaveChanges();
+
+            return sighting;
```
Comment (TimLeach635) [inline backend/Repositories/SightingRepo.cs:47]: You shouldn't return the deleted object, it's been deleted! From the point of view of the API, this Sighting should no longer exist!

File: backend/Repositories/SightingRepo.cs [hunk 3]
```diff
@@ -7,6 +8,8 @@ namespace WhaleSpotting.Repositories
     public interface ISightingRepo
     {
         IEnumerable<Sighting> GetAllSightings();
+        Sighting GetById(int id);
+        Sighting DeleteSighting(int id);
```
Comment (TimLeach635) [inline backend/Repositories/SightingRepo.cs:12]: As below, a deletion shouldn't return anything

File: backend/Services/SightingService.cs [hunk 4]
```diff
@@ -22,5 +23,10 @@ public IEnumerable<Sighting> GetAllSightings()
         {
             return _sightingRepo.GetAllSightings();
         }
+
+        public Sighting DeleteSighting(int id)
```
Comment (TimLeach635) [inline backend/Services/SightingService.cs:27]: Deletions shouldn't return anything!

Comment (TimLeach635) [reply backend/Services/SightingService.cs:27]: Also, rename this to match the repo

File: backend/Repositories/SightingRepo.cs [hunk 5]
```diff
@@ -24,5 +27,25 @@ public IEnumerable<Sighting> GetAllSightings()
                 .Sightings
                 .Include(s => s.Species);
         }
+
+         public Sighting GetById(int id)
+        {
+            return _context
+                .Sightings
+                .Single(s => s.Id == id);
+        }
```
Comment (TimLeach635) [inline backend/Repositories/SightingRepo.cs:36]: Actually, why did you create this method and then not use it? If you're not using it, please delete it!

**Review by TimLeach635 — CHANGES_REQUESTED** [review]
A few changes here!

File: backend/Repositories/SightingRepo.cs [hunk 6]
```diff
@@ -24,5 +27,25 @@ public IEnumerable<Sighting> GetAllSightings()
...
+         public Sighting DeleteSighting(int id)
+        {
+             Sighting sighting = _context
+                .Sightings
+                .Include(s => s.Species)
+                .Single(s => s.Id == id);
```
[the remark this answers is not in the document]
Comment (Jordan-Renaud) [reply backend/Repositories/SightingRepo.cs:42]: Do I need to include the species in this at all to update the IsDeleted property?

[the remark this answers is not in the document]
Comment (TimLeach635) [reply backend/Repositories/SightingRepo.cs:42]: Very good point, no you don't!

**Review by TimLeach635 — APPROVED** [review]

**Pull request merged**